src/libraries/CurveLib.sol

```diff
@@ -0,0 +1,254 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.21;
+
+import { IERC20 }   from "forge-std/interfaces/IERC20.sol";
+import { AccessControl } from "openzeppelin-contracts/contracts/access/AccessControl.sol";
+
+import { IRateLimits } from "../interfaces/IRateLimits.sol";
+import { IALMProxy } from "../interfaces/IALMProxy.sol";
+
+import { RateLimitHelpers } from "../RateLimitHelpers.sol";
+import { Types }            from "../Types.sol";
+
+interface ICurvePoolLike is IERC20 {
+    function add_liquidity(
+        uint256[] memory amounts,
+        uint256 minMintAmount,
+        address receiver
+    ) external;
+    function balances(uint256 index) external view returns (uint256);
+    function coins(uint256 index) external returns (address);
+    function exchange(
+        int128  inputIndex,
+        int128  outputIndex,
+        uint256 amountIn,
+        uint256 minAmountOut,
+        address receiver
+    ) external returns (uint256 tokensOut);
+    function get_virtual_price() external view returns (uint256);
+    function N_COINS() external view returns (uint256);
+    function remove_liquidity(
+        uint256 burnAmount,
+        uint256[] memory minAmounts,
+        address receiver
+    ) external;
+    function stored_rates() external view returns (uint256[] memory);
+}
+
+library CurveLib {
+    
+    bytes32 public constant LIMIT_CURVE_DEPOSIT        = keccak256("LIMIT_CURVE_DEPOSIT");
+    bytes32 public constant LIMIT_CURVE_SWAP           = keccak256("LIMIT_CURVE_SWAP");
+    bytes32 public constant LIMIT_CURVE_WITHDRAW       = keccak256("LIMIT_CURVE_WITHDRAW");
```

Comment:
Extra spaces

Reply:
u meant the spaces above these lines right